ur_robot_driver/README.md

```diff
@@ -4,6 +4,74 @@ This package contains the actual driver for UR robots. It is part of the *univer
 repository and requires other packages from that repository. Also, see the [main repository's
 README](../README.md) for information on how to install and startup this driver.
 
+
+## About this branch
+
+This branch is a beta test of new interfaces towards the robot.
+It is, as such, to be considered under development, and documentation may in parts be missing and
+things might change in the near future.
+It is required to use the beta-testing branch of the [Universal Robots Client Library](https://github.com/UniversalRobots/Universal_Robots_Client_Library). 
+
+Additionally, the packages in the [cartesian ros control
+repository](https://github.com/fzi-forschungszentrum-informatik/cartesian_ros_control) are needed,
+both for the provided cartesian interface definitions and new controllers utilizing the new
+interfaces to the robot.
+
+### Features
+
+This branch introduces several new interfaces via which to communicate with the robot.
+
+* Cartesian pose streaming
+* Cartesian velocity streaming
+* Joint-based trajectory forwarding
+* Cartesian trajectory forwarding
+* Cartesian control using the joint-based robot interface
+
+All of them can be used with corresponding ROS controllers.
+More information about the controllers can be found in the cartesian ros control repository.
+
+##### Cartesian pose streaming
+
+Start the preloaded `pose_traj_controller`.
+This controller allows ROS-level interpolation and streaming of cartesian poses to the robot.
+It uses the newly created FollowCartesianTrajectory action.
+
+##### Cartesian velocity streaming
+
+Start the preloaded `twist_controller`.
+This controller allows for streaming twist commands via the `twist_controller/command` topic to be
+interpreted as cartesian velocity control commands by the robot.
+
+##### Joint-based trajectory forwarding
+
+Start the preloaded `forward_joint_traj_controller`.
+This controllers allows for forwarding full trajectories to the robot, to be interpolated and
+executed completely by the internal robot control.
+While it uses the same FollowJointTrajectory interface as the previously existing trajectory
+control, during execution no control commands other than a keepalive signal are sent to the robot.
+A cancellation of the trajectory is still possible.
+
+Of special not for the forwarded trajectory is that, currently, blending is activated on the robot.
+This produces smoother trajectories, but causes target points other than the last to not
+necessarily be reached.
+One of the objectives of this beta test is to gather feedback for this behaviour and look into
+feasible ways to control this blending radius, that has to directly corresponding field in the ROS
+trajectory interface definition, to be set based on the commanded trajectory.
```

Comment:
What about using the `path_tolerance` in some way?

Reply:
Yes, the path_tolerance should most likely be the basis for this - we do have to move from joint based to position(/velocity/acceleration) tolerances to a cartesian radius, though.
As this transformation is dependent on the robot configuration, there are some decisions on complexity vs. accuracy to be made here - especially thinking about transparency for the person setting these path_tolerance values.

Reply:
This is discussing the `forward_joint_traj_controller`is it not? Or all forwarding controllers?

Reply:
That would concern both forwarding controllers. There are two types, one for joint-based trajectories and one for Cartesian-based trajectories.

Reply:
Yes, at least primarily. For the cartesian forwarding controller using the position error in the cartesian tolerance to calculate a good blend radius should be trivial.